File: ARRAYS/Easy/Two Sum/Code1.cpp
/*
PROBLEM LINK:- https://leetcode.com/problems/two-sum/
*/

class Solution {
public:
    vector<int> twoSum(vector<int>& v, int x) {
        int n = v.size();
        vector<int> r;
        for(int i = 0; i < n; i++)
        {
            for(int j = i+1; j < n; j++)
            {
                if(i != j and v[i]+v[j] == x)
                {
                    r.push_back(i);
                    r.push_back(j);
                }
            }
        }
        return r;
    }
};

File: TREE/EASY/Minimum Depth of Binary Tree/Code1.py
# PROBLEM LINK:- https://leetcode.com/problems/minimum-depth-of-binary-tree/

class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        
        l = self.minDepth(root.left)
        r = self.minDepth(root.right)
        
        if l == 0:
            return 1 + r
        elif r == 0:
            return 1 + l
        return 1 + min(l, r)
